### PR TITLE
[test] Add unit test for AvroCollectionElementComparator

### PR DESCRIPTION
**Pull request by sushantmane**
## How was this PR tested?

Internal CI

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-client-common/src/test/java/com/linkedin/venice/schema/merge/AvroCollectionElementComparatorTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  @Test
+  public void testCompareWhenSchemaTypeIsNullShouldReturnZero() {
+    assertEquals(INSTANCE.compare(new GenericData(), new GenericData(), Schema.create(Schema.Type.NULL)), 0);
+  }
+
+  @Test
+  public void testCompareWithStringSchema() {
+    assertEquals(INSTANCE.compare("A", "A", Schema.create(Schema.Type.STRING)), 0);
+    assertTrue(INSTANCE.compare("A", "B", Schema.create(Schema.Type.STRING)) < 0);
+    assertTrue(INSTANCE.compare("Z", "A", Schema.create(Schema.Type.STRING)) > 1);
```
Comment (ZacAttack) [inline internal/venice-client-common/src/test/java/com/linkedin/venice/schema/merge/AvroCollectionElementComparatorTest.java:27]: 0 1 and -1 are the outputs.  > 1 is kind of confusing.

Comment (sushantmane) [reply internal/venice-client-common/src/test/java/com/linkedin/venice/schema/merge/AvroCollectionElementComparatorTest.java:27]: For `String` types Avro does byte-by-byte comparison
Here is the doc string for the compare function:`Lexicographically compare bytes. If equal, return zero. If greater-than, return a positive value, if less than return a negative value.`

**Review by ZacAttack — DISMISSED** [review]
Looks good, Thanks for adding!

File: internal/venice-client-common/src/test/java/com/linkedin/venice/schema/merge/AvroCollectionElementComparatorTest.java (hunk 1 above)
Comment (sushantmane) [reply internal/venice-client-common/src/test/java/com/linkedin/venice/schema/merge/AvroCollectionElementComparatorTest.java:27]: Updated to use <, ==, >

**Comment by sushantmane** [comment]
Thanks for the review, Zac!

**Review by ZacAttack — APPROVED** [review]

**Pull request merged**